### DISCUSSION TITLE
Permission to login root terminal by MeshCentral user

### DISCUSSION THREAD

**Discussion by cheuklam** [Ylianst/MeshCentral#6337] [General]
Hihi, I am super happy with MeshCentral and have been using it for quite a while, and until today I just realize I can right click on the connect button and choose User Shell in terminal connection.

This makes me think, if it is possible for me to limit MeshCentral logged in user to use root shell? e.g. ITAdmin can use root shell, while ITStaff can only use user shell (would be even better if I can control which user shell). This make MeshCentral a strong candidate to replace Zero Trust solution!

Thanks in advance.

**Comment by si458** [comment]
Right now the isn't a way to set this per user/per group
BUT you can set it for the whole server and every domain
![Screenshot_20240824_122229_Firefox.jpg](https://github.com/user-attachments/assets/ac57f090-49c2-40f4-a524-42cf022ed3d0)

Inside of config.json you set `settings` then u set `terminal` then u set `linuxshell` and set it to either root, user or login

Then when a user connects they drop it to either a root session, the current logged in users session OR they get a login prompt so they can login manually!

Remember this is for linux machines only

**Reply by kubamaruszczak** [reply]
@si458 did anything change in that manner? Is it now possible to restrict certain users to only have `login` option, while admin users have access to `root` terminal?
